frontend/src/console/components/CustomPages/CustomPages.tsx

```diff
@@ -0,0 +1,218 @@
+import * as React from 'react';
+import './styles.scss';
+import {
+  ConsolePage,
+  ConsolePageCustomizationContainer
+} from 'console/components';
+import { LMS_CUSTOM_PAGE_LINK_MAP, TINYMCE_API_KEY } from 'global/constants';
+import { InstancesModel } from 'console/models';
+import { connect } from 'react-redux';
+import { RootState } from 'global/state';
+import { WrappedMessage } from 'utils/intl';
+import { updateStaticContentOverridesFieldValue } from 'console/actions';
+import { capitalizeFirstLetter } from 'utils/string_utils';
+import { StaticContentOverrides } from 'ocim-client';
+import { Row, Col, Form } from 'react-bootstrap';
+import { Prompt } from 'react-router';
+import { Editor } from '@tinymce/tinymce-react';
+import messages from './displayMessages';
+
+interface State {
+  [key: string]: any;
+  pageContent: any;
+}
+interface ActionProps {
+  updateStaticContentOverridesFieldValue: Function;
+}
+interface StateProps extends InstancesModel {}
+interface Props extends StateProps, ActionProps {
+  // Passing URL as parameter to custom page
+  match: {
+    params: {
+      pageName: string;
+    };
+  };
+}
+
+export class CustomPagesComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      pageContent: this.getPageContentFromState()
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
```

Comment:
Shouldn't we define it like `componentDidUpdate(prevProps: Props) {` to overload the parent's class method?

Reply:
It binds in the same way afaik, so I went with the `= () =>` to keep all implementations consistent inside this component.

Reply:
Do you want me to change it to `componentDidUpdate(prevProps: Props) {`?

Reply:
Ah, I see that my IDE mistakenly took the TypeScript version from `api_client` `v2.9.2` instead of the base one `v3.6.4`, which resulted in [this error](https://github.com/microsoft/TypeScript/issues/17113). It's fine to keep it this way. Sorry for the confusion!